jupyter-org-client.el

```diff
@@ -987,6 +990,30 @@ is a stream result. Otherwise return nil."
     (when (looking-at-p "\\(?::[\t ]\\|#\\+END_EXAMPLE\\)")
       (line-end-position (unless (eq (char-after) ?:) 0)))))
 
+(defun jupyter-org--find-mime-types (req-types)
+  "Return the keywords in `jupyter-org-mime-types' that match REQ-TYPES.
+
+If a match is not found, return nil. Try to be intelligent and
+return what the user might intend to use.
+
+REQ-TYPES can be a strings such as `plain', `plain html' or
+`text/plain'. The string `text' is translated to `:text/plain'
+and `image' to `:image/png'."
+  (and req-types ;; proceed only if REQ-TYPES is not nil
+       ;; iterate every specified REQ-TYPE looking for matches in MIME-TYPES
+       (mapcar (lambda (req-type)
```

Comment:
This function can either return `nil`, a list like `(:text/plain :image/png)`, or a list like `(:text/plain :image/png nil)`. It doesn't look like the last form with `nil` in the list is being considered anywhere though.

If a user puts in an invalid mimetype in the `:display` property there will be a `nil` in the returned list. Maybe we should verify that the `:display` property is valid in `org-babel-execute:jupyter`? I don't think it would be a good idea to do any error handling in this function since this is called after the code block has been evaluated.

Reply:
The does not return now "nil" if the mimetype is not known. Instead, it is just ignored.